cloudify_aws/common/tests/test_iface_requirement.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import yaml
+from mock import patch, MagicMock
+from importlib import import_module
+
+from cloudify.state import current_ctx
+from cloudify.exceptions import (OperationRetry, NonRecoverableError)
+
+from cloudify_aws.common.tests.test_base import TestBase
+
+REL_LIFE = 'cloudify.interfaces.relationship_lifecycle'
+
+
+def get_callable(operation_mapping):
+    if not isinstance(operation_mapping, dict):
+        raise Exception(
+            'Operation {op} is not dict.'.format(op=operation_mapping))
+    elif 'implementation' not in operation_mapping:
+        return
+    elif operation_mapping['implementation'] == '~':
+        return
+    elif not operation_mapping['implementation']:
+        return
+    modules = operation_mapping['implementation'].split('.')
+    del modules[0]
+    func = modules.pop(-1)
+    import_path = '.'.join(modules)
+    module = import_module(import_path)
+    return module and getattr(module, func, None)
+
+
+class testIfaceRequirement(TestBase):
+
+    @staticmethod
+    def get_plugin_yaml():
+        plugin_yaml_path = os.path.abspath(
+            os.path.join(
+                os.path.dirname(__file__),
+                '..', '..', '..', 'plugin.yaml'))
+        plugin_yaml_file = open(plugin_yaml_path, 'r')
+        return yaml.load(plugin_yaml_file, Loader=yaml.FullLoader)
+
+    @staticmethod
+    def get_node_type_operations(plugin_yaml):
+        operations = []
+        for _, node in plugin_yaml['node_types'].items():
+            try:
+                op_list = node['interfaces']['cloudify.interfaces.lifecycle']
+            except KeyError:
+                continue
+            for __, op in op_list.items():
+                module = get_callable(op)
+                operations.append(module)
+        return operations
+
+    @staticmethod
+    def get_relationships_operations(plugin_yaml):
+        operations = []
+        for _, rel in plugin_yaml['relationships'].items():
+            op_list = rel.get('source_interfaces', {}).get(REL_LIFE, {})
+            op_list.update(rel.get('target_interfaces', {}).get(REL_LIFE, {}))
+            for __, op in op_list.items():
+                module = get_callable(op)
+                operations.append(module)
+        return operations
+
+    def get_op_ctx(self, operation):
+        mock_group = MagicMock()
+        mock_group.type_hierarchy = [
+            'cloudify.relationships.depends_on',
+            'cloudify.relationships.contained_in'
+        ]
+        mock_group.target.instance.runtime_properties = {
+            'aws_resource_id': 'aws_id',
+            'aws_resource_arn': 'foo',
+            'ListenerArn': 'arn:aws:foo',
+            'resource_config': {}
+        }
+        mock_group.target.node.type_hierarchy = [
```

Comment:
Why you choose these specific nodes?

Reply:
These are just belonging to nodes that use them in their operations.